### PR TITLE
[v12] Update gravitational/license module

### PR DESCRIPTION
**Pull request by zmb3**
Backport #36117 to branch/v12

### CODE REVIEW

**Review by strideynet — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Review by AntonAM — APPROVED** [review]

**Comment by zmb3** [comment]
tbot test failure was due to a clockwork update.

Nic fixed it in master in https://github.com/gravitational/teleport/pull/24099 but that fix never made it into branch/v12.

**Comment by zmb3** [comment]
@rosstimothy can you double check the dependency changes in here and see if anything scares you? gravitational/license didn't used to be a proper go module, and now that it is it declares a bunch of newer dependencies which means a lot of things want to be bumped here.

**Review by rosstimothy — COMMENTED** [review]
I've not gone through the changelog of every update but most of the usual suspects that cause problems don't appear to be updated. Is this really needed so close to EOL for v12?

**Comment by zmb3** [comment]
> I've not gone through the changelog of every update but most of the usual suspects that cause problems don't appear to be updated. Is this really needed so close to EOL for v12?

Somewhat. We need this in order to properly bill self-hosted customers, and as you know they often lag behind..

File: go.mod [hunk 1]
```diff
@@ -61,16 +61,15 @@ require (
...
 	github.com/google/go-attestation v0.4.3
 	github.com/google/go-cmp v0.6.0
 	github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
-	github.com/google/uuid v1.3.1
-	github.com/googleapis/gax-go/v2 v2.11.0
+	github.com/google/uuid v1.4.0
+	github.com/googleapis/gax-go/v2 v2.12.0
 	github.com/gorilla/websocket v1.5.0
 	github.com/gravitational/form v0.0.0-20151109031454-c4048f792f70
-	github.com/gravitational/license v0.0.0-20210218173955-6d8fb49b117a
```
[the remark this answers is not in the document]
Comment (zmb3) [reply go.mod:73]: I must have done something wrong. Let me try again.

**Comment by zmb3** [comment]
Abandoning v12 for now, as the license update also wants to upgrade clockwork which breaks too many tests.
